dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/BitstreamRestRepository.java

```diff
@@ -135,6 +155,76 @@ protected void delete(Context context, UUID id) throws AuthorizeException {
         }
     }
 
+    /**
+     * Find the bitstream for the provided handle and sequence or filename.
+     * When a bitstream can be found with the sequence ID it will be returned if the user has "METADATA_READ" access.
+     *
+     * @param handle    The handle of the item
+     * @param sequence  The sequence ID of the bitstream
+     * @param filename  The filename of the bitstream
+     *
+     * @return a Page of BitstreamRest instance matching the user query
+     */
+    @SearchRestMethod(name = "byItemHandle")
+    public BitstreamRest findByItemHandle(@Parameter(value = "handle", required = true) String handle,
+                                          @Parameter(value = "sequence") Integer sequence,
+                                          @Parameter(value = "filename") String filename) {
+        if (StringUtils.isBlank(filename) && sequence == null) {
+            throw new IllegalArgumentException("The request should include a sequence or a filename");
+        }
+
+        try {
+            Context context = obtainContext();
+            DSpaceObject dSpaceObject = handleService.resolveToObject(context, handle);
+
+            if (!(dSpaceObject instanceof Item)) {
+                throw new UnprocessableEntityException("The provided handle does not correspond to an existing item");
+            }
+            Item item = (Item) dSpaceObject;
+
+            List<Bitstream> matchedBitstreams = getMatchedBitstreams(item, sequence, filename);
+
+            if (matchedBitstreams.isEmpty()) {
+                return null;
+            }
+
+            for (Bitstream bitstream : matchedBitstreams) {
+                if (bitstreamMetadataReadPermissionEvaluatorPlugin
```

Comment:
If the first bitstream matching (on bitstreamName, there could be multiple bitstreams with the requested name), is not available to requesting user, checked here, then the next matching bitstream could still be returned. If for example there  are two bitstreams on an item, with the same name, one restricted and one not, we still want the user to be able to use this endpoint to retrieve the non-restricted one.

Reply:
Maybe we could just add a comment above this `for` to explain it better?  I agree with the logic here though.... this endpoint *might* match multiple bitstreams. In that scenario we should return the first Bitstream where you have read permissions.

(This is a slight change in the logic from 6.x, as I believe we just returned the first bitstream that matched in 6.x, regardless of your permissions. But, I think the permissions check is an improvement.)

Reply:
I'm still convinced that is better to avoid this direct dependency. This endpoint is only needed to provide support for old URL and in the previous version only the first retrieved bitstream regardless to the security is considered. So we can keep the same behavior here avoiding to make this extra check. If the first bitstream is invalid (the user is not allowed to access it) I expect that the converter will hide it and the endpoint will return no bitstream, this is close to what would happen on an old version where the user will hit a permission issue

Reply:
@MarieVerdonck : I'm OK with @abollini 's suggestion to just remove the permission check for now.  As it didn't exist in 6.x, and this endpoint is really only about redirecting old 6.x URLs to new 7.x URLs, we can just remove it.  If we find a need for it later, we can add it back then.

If possible, I'd like to get this moved forward soon, and it sounds like this is the last remaining feedback from @abollini, so hopefully we can just get this addressed and merged soon.